File: testdata/fixtures/ruby/CWE-798.rb
# CWE-798 Hardcoded Credentials in Ruby
# Embedding passwords, secrets or API keys directly in source code exposes
# them to anyone with repository access and makes rotation impossible.

# ruleid: ruby-hardcoded-credential
password = "supersecret123"

# ruleid: ruby-hardcoded-credential
api_key = "AKIAIOSFODNN7EXAMPLE"

# ruleid: ruby-hardcoded-credential
secret = "my_hardcoded_secret_!@#"

# ruleid: ruby-hardcoded-credential
aws_secret = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"

# ok: ruby-hardcoded-credential
password = ENV["PASSWORD"]

# ok: ruby-hardcoded-credential
api_key = ENV.fetch("API_KEY")

# ok: ruby-hardcoded-credential
secret = get_secret_from_vault()

File: testdata/fixtures/clojure/CWE-798.clj
;; CWE-798: Hardcoded
(ns fixture.cwe798)
;; ruleid: clojure-hardcoded
(def password "hunter2")
;; ruleid: clojure-hardcoded
(def api-key "ak_s3cr3t123")
;; ruleid: clojure-hardcoded
(def secret-token "t0k3n123")
;; ok: clojure-hardcoded
(def password "password")
;; ok: clojure-hardcoded
(def username "hunter2")
;; ok: clojure-hardcoded
(def api-key "API_KEY")
